### PR TITLE
1268491: RestEasy 3 changed behavior when a JAX-RS resource's paramet…

### PR DESCRIPTION
**Pull request by nguyenfilip**

### CODE REVIEW

File: common/src/main/java/org/candlepin/common/exceptions/mappers/BadQueryParameterHandler.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * for that exception.
+ * 
+ * Historical notes: This had to be extracted from BadRequestExceptionMapper, because the new
+ * RestEasy changed the way it handles (now throws NotFoundException)
+ * unparseable parameters of JAX-RS resources.
+ * 
+ * @author fnguyen
+ *
+ */
+public class BadQueryParameterHandler {//extends CandlepinExceptionMapper {
```
Comment (barnabycourt) [inline common/src/main/java/org/candlepin/common/exceptions/mappers/BadQueryParameterHandler.java]: If this no longer extends CandlepinExceptionMapper then this comment should be removed

File: common/src/main/java/org/candlepin/common/exceptions/mappers/BadQueryParameterHandler.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		}
+
+		return false;
+	}
+
+	/**
+	 * Builds response for this exception. 
+	 * 
+	 * @param exception
+	 * @return
```
Comment (barnabycourt) [inline common/src/main/java/org/candlepin/common/exceptions/mappers/BadQueryParameterHandler.java]: This javadoc is missing information about the details of what this method does, the expected parameter, and what it returns.

File: common/src/main/java/org/candlepin/common/exceptions/mappers/BadRequestExceptionMapper.java [hunk 3]
```diff
@@ -39,47 +41,23 @@
...
+
+	@Override
+	public Response toResponse(BadRequestException exception) {
+		if (badQueryParamHandler.canHandle(exception)) {
+			return badQueryParamHandler.getResponse(exception);
+		} else {
+			// This is here to preserve old functionality. 
+			// It may be just as good to use getDefaultBuilder here
+			Map<String, String> map = VersionUtil.getVersionMap();
+	        ResponseBuilder bldr = Response.status(Status.BAD_REQUEST).type(
```
Comment (barnabycourt) [inline common/src/main/java/org/candlepin/common/exceptions/mappers/BadRequestExceptionMapper.java]: indentation for this block is not consistent

**Comment by Ceiu** [comment]
retest this please

**Comment by nguyenfilip** [comment]
Amending to clear up checkstyle problems related to files modified by this PR. Also addressing @barnabycourt  comments.

**Comment by nguyenfilip** [comment]
Based on @awood note about usage of & I decided to cover this with a bit more unit tests. Do not merge yet please.

**Comment by nguyenfilip** [comment]
retest this please

**Comment by awood** [comment]
Looks good to me.  I will merge when you are ready.

**Comment by nguyenfilip** [comment]
Good, its ready to be merged, thanks @awood  for all the feedback

**Pull request merged**